src/test/java/com/example/petbuddybackend/controller/CaretakerControllerTest.java

```diff
@@ -0,0 +1,72 @@
+package com.example.petbuddybackend.controller;
+
+import com.example.petbuddybackend.dto.user.AccountDataDTO;
+import com.example.petbuddybackend.dto.user.CaretakerDTO;
+import com.example.petbuddybackend.service.user.CaretakerService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.MockitoAnnotations;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.http.MediaType;
+import org.springframework.security.oauth2.jwt.JwtDecoder;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.util.List;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+
+@SpringBootTest
+@AutoConfigureMockMvc
+public class CaretakerControllerTest {
+
+    @MockBean
+    private CaretakerService caretakerService;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private JwtDecoder jwtDecoder;
+
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    void getCaretakers_shouldReturnFilteredResults() throws Exception {
+        var caretakerDTO1 = CaretakerDTO.builder()
```

Comment:
Wydzielilbym osobna metode na robienie Caretakera, bo wraz z przytostem testow bedzie sie powiekszalo na nich zapotrzebowanie

Reply:
w sumie to chyba jest tylko nie uzyto tutaj

Reply:
To DTO tylko tutaj tak tworzę i nie potrzebowałem tego bardzo rozbudowanego, tylko "name". Zostawiam to na razie